src/Microsoft.IdentityModel.Protocols.PoP/SignedHttpRequest/SignedHttpRequestCreationData.cs

```diff
@@ -0,0 +1,92 @@
+    /// </summary>
+    public class SignedHttpRequestCreationData : SignedHttpRequestData
+    {
+        /// <summary>
+        /// Initializes a new instance of <see cref="SignedHttpRequestCreationData"/>.
+        /// </summary>
+        /// <remarks>
+        /// <paramref name="accessToken"/> has to contain the 'cnf' claim so that PoP key can be resolved on the validation side.
+        /// https://tools.ietf.org/html/rfc7800#section-3.1
+        /// Default <see cref="CallContext"/> is created.
+        /// </remarks>
+        /// <param name="accessToken">An access token that contains the 'cnf' claim.</param>
+        /// <param name="httpRequestData">A structure that represents an outgoing http request.</param>
+        /// <param name="httpRequestSigningCredentials">A security key and algorithm that will be used to sign the (Signed)HttpRequest.</param>
+        /// <param name="signedHttpRequestCreationPolicy">A policy for creating the SignedHttpRequest.</param>
+        public SignedHttpRequestCreationData(string accessToken, HttpRequestData httpRequestData, SigningCredentials httpRequestSigningCredentials, SignedHttpRequestCreationPolicy signedHttpRequestCreationPolicy) 
+            : base(httpRequestData, CallContext.Default)
+        {
+            AccessToken = !string.IsNullOrEmpty(accessToken) ? accessToken : throw LogHelper.LogArgumentNullException(nameof(accessToken));
+            HttpRequestSigningCredentials = httpRequestSigningCredentials ?? throw LogHelper.LogArgumentNullException(nameof(httpRequestSigningCredentials));
+            SignedHttpRequestCreationPolicy = signedHttpRequestCreationPolicy ?? throw LogHelper.LogArgumentNullException(nameof(signedHttpRequestCreationPolicy));
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="SignedHttpRequestCreationData"/>.
+        /// </summary>
+        /// <remarks>
+        /// <paramref name="accessToken"/> has to contain the 'cnf' claim so that PoP key can be resolved on the validation side.
+        /// https://tools.ietf.org/html/rfc7800#section-3.1
+        /// </remarks> 
+        /// <param name="accessToken">An access token that contains the 'cnf' claim.</param>
+        /// <param name="httpRequestData">A structure that represents an outgoing http request.</param>
+        /// <param name="httpRequestSigningCredentials">Defines the security key and algorithm that will be used to sign the (Signed)HttpRequest.</param>
+        /// <param name="signedHttpRequestCreationPolicy">A policy for creating the SignedHttpRequest.</param>
+        /// <param name="callContext">An opaque context used to store work when working with authentication artifacts.</param> 
+        public SignedHttpRequestCreationData(string accessToken, HttpRequestData httpRequestData, SigningCredentials httpRequestSigningCredentials, SignedHttpRequestCreationPolicy signedHttpRequestCreationPolicy, CallContext callContext) 
```

Comment:
transferred comment:
@brentschmaltz:
> did you consider the name PopTokenDescriptor, this would be in line with SecurityTokenDescriptor.
it has similar semantics.

Reply:
@brentschmaltz  _SecurityTokenDescriptor_ is used only when creating a token. I needed classes to hold data necessary for both signed http request creation and validation.

Reply:
Renamed to SignedHttpRequestDescriptor.